obelisk/cpp/zoo/hardware/sensing/zed/zed2_sensors_interface.h

```diff
@@ -0,0 +1,420 @@
+                height_                                   = 1080;
+                width_                                    = 1920;
+            } else if (resolution == "2k") {
+                cam_param_dicts_[cam_index]["resolution"] = "2K";
+                resolution_                               = sl::RESOLUTION::HD2K;
+                height_                                   = 1242;
+                width_                                    = 2208;
+            } else {
+                std::string err_msg = "Invalid resolution for camera " + std::to_string(cam_index) + " in " +
+                                      params_path + "! Must be one of 'VGA', '720', '1080', '2K'.";
+                RCLCPP_ERROR(this->get_logger(), err_msg.c_str());
+                throw std::runtime_error(err_msg);
+            }
+            cam_param_dicts_[cam_index]["resolution"] = resolution;
+
+            // Check fps
+            int fps = cam_param_dict["fps"] ? cam_param_dict["fps"].as<int>() : 0;
+            if (fps == 0) {
+                if (resolution == "vga")
+                    fps = 100;
+                else if (resolution == "720")
+                    fps = 60;
+                else if (resolution == "1080")
+                    fps = 30;
+                else if (resolution == "2k")
+                    fps = 15;
+            }
+            if (fps != 15 && fps != 30 && fps != 60 && fps != 100) {
+                std::string err_msg = "Invalid fps for camera " + std::to_string(cam_index) + " in " + params_path +
+                                      "! Must be one of 15, 30, 60, 100.";
+                RCLCPP_ERROR(this->get_logger(), err_msg.c_str());
+                throw std::runtime_error(err_msg);
+            }
+            cam_param_dicts_[cam_index]["fps"] = std::to_string(fps);
+            fps_                               = fps;
+
+            // Check depth
+            bool depth                           = cam_param_dict["depth"] ? cam_param_dict["depth"].as<bool>() : false;
+            cam_param_dicts_[cam_index]["depth"] = depth ? "true" : "false";
+            depth_                               = depth;
+
+            // Check side
+            std::string side = cam_param_dict["side"] ? cam_param_dict["side"].as<std::string>() : "left";
+            std::transform(side.begin(), side.end(), side.begin(), ::tolower);
+            if (side != "left" && side != "right") {
+                std::string err_msg = "Invalid side for camera " + std::to_string(cam_index) + " in " + params_path +
+                                      "! Must be one of 'left', 'right'.";
+                RCLCPP_ERROR(this->get_logger(), err_msg.c_str());
+                throw std::runtime_error(err_msg);
+            }
+            cam_param_dicts_[cam_index]["side"] = side;
+        }
+
+        // Check that all cameras have the same resolution, fps, and depth setting
+        std::set<std::string> resolutions, fpss, depths;
```

Comment:
Not sure why this is a std::set and not another container. I've never used set before, but it looks like its useful for sorting and idk if you are taking advantage of that.

Reply:
This was a holdover that got translated from some python logic (which tbf also didn't need to bust out sets). I've refactored it.